integration_tests/reload_api_test.go

```diff
@@ -31,6 +33,28 @@ var _ = Describe("reload API endpoint", func() {
 			Expect(resp.StatusCode).To(Equal(405))
 			Expect(resp.Header.Get("Allow")).To(Equal("POST"))
 		})
+
+		It("eventually reloads the routes", func() {
+			addRoute("/foo", NewRedirectRoute("/qux", "prefix"))
+
+			start := time.Now()
+			doRequest(newRequest("POST", routerAPIURL("/reload")))
+			end := time.Now()
+			duration := end.Sub(start)
+
+			Expect(duration.Nanoseconds()).To(BeNumerically("<", 1000000))
+
+			addRoute("/bar", NewRedirectRoute("/qux", "prefix"))
+			doRequest(newRequest("POST", routerAPIURL("/reload")))
+
+			time.Sleep(time.Second * 1)
```

Comment:
Minor detail: does this need to sleep for a full second?  If not, it worth shortening it to avoid making the tests slower than they need to be.